src/RemoteInboxNotifications/SpecRunner.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * Runs a single spec.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+/**
+ * Runs a single spec.
+ */
+class SpecRunner {
+	/**
+	 * Run the spec.
+	 *
+	 * @param object $spec The spec to run.
+	 */
+	public static function run_spec( $spec ) {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// Create or update the note.
+		$existing_note_ids = $data_store->get_notes_with_name( $spec->slug );
+		if ( 0 === count( $existing_note_ids ) ) {
+			$note = new WC_Admin_Note();
+			$note->set_status( WC_Admin_Note::E_WC_ADMIN_NOTE_PENDING );
+		} else {
+			$note = new WC_Admin_Note( $existing_note_ids[0] );
+		}
+
+		// Evaluate the spec and get the new note status.
+		$rule_evaluator = new RuleEvaluator(
+			new GetRuleProcessor()
+		);
+
+		$previous_status = $note->get_status();
+		$status          = EvaluateAndGetStatus::evaluate(
+			$spec,
+			$previous_status,
+			$rule_evaluator
+		);
+
+		// If the status is changing, update the created date to now.
+		if ( $previous_status !== $status ) {
+			$note->set_date_created( time() );
+		}
+
+		// Get the matching locale or fall back to en-US.
+		$locale = self::get_locale( $spec->locales );
+
+		if ( null === $locale ) {
+			return;
+		}
+
+		// Set up the note.
+		$note->set_title( $locale->title );
+		$note->set_content( $locale->content );
+		$note->set_content_data( (object) array() );
+		$note->set_status( $status );
+		$note->set_type( $spec->type );
+		$note->set_icon( $spec->icon );
```

Comment:
With the new notes designs being merged in, `set_icon` has been deprecated so we should remove this to not get warnings in the woo log files on sites.

Reply:
Done in a68f2a4b